src/routes/[deviceUID]/dashboard/MapboxMap.svelte

```diff
@@ -0,0 +1,137 @@
+<script lang="ts">
+	import { onMount } from 'svelte';
+	import { fade } from 'svelte/transition';
+	import { format, parseISO } from 'date-fns';
+	import { aqiColors, aqiRanges, getAQIColor } from '$lib/services/air';
+	import { DATE_TIME_KEY } from '$lib/constants';
+	import mapboxgl from 'mapbox-gl';
+	import type { AirnoteReading } from '$lib/services/AirReadingModel';
+
+	export let lastReading: AirnoteReading;
+
+	let map;
+	let airnoteMarker;
+	let markerColor;
+	let zoom = 10;
+	let popup;
+	let mapboxToken =
+		'pk.eyJ1IjoicGFpZ2VuMTEiLCJhIjoiY2lyemJlZ3A0MDBqZTJ5cGs5ZHJicjI2YyJ9.2-dZqM-k2obDN47BpWq5Lw';
+
+	onMount(() => {
+		mapboxgl.accessToken = mapboxToken;
+		map = new mapboxgl.Map({
+			container: 'map',
+			style: 'mapbox://styles/mapbox/streets-v11',
+			center: [lastReading.lon, lastReading.lat],
+			zoom
+		});
+
+		aqiRanges.forEach((layer, i: number) => {
+			const color = aqiColors[i];
+			const item = document.createElement('div');
+			const key = document.createElement('span');
+			const legend: HTMLElement | null = document.getElementById('legend');
+			item.className = 'legend-wrapper';
+			item.style.display = 'flex';
+			item.style.alignItems = 'center';
+			key.className = 'legend-key';
+			key.style.background = color;
+			key.style.height = '15px';
+			key.style.width = '15px';
+			key.style.marginRight = '8px';
+
+			const value = document.createElement('span');
+			value.innerHTML = `${layer}`;
+			item.appendChild(key);
+			item.appendChild(value);
+			if (legend) {
+				legend.appendChild(item);
+			}
+		});
+
+		const d = parseISO(lastReading.captured);
+
+		popup = new mapboxgl.Popup().setHTML(
+			`<p style="text-align: center; margin:0">${format(d, DATE_TIME_KEY)}</p>
+      <h3 style="margin: 0">Device ID: ${lastReading.device_uid}</h3>
+      <p style="margin: 0; display: flex; justify-content: space-between"><span class="mapboxgl-pm">AQI:</span> ${Math.round(
+				lastReading.aqi
+			)}</p>
+      <p style="margin: 0; display: flex; justify-content: space-between"><span class="mapboxgl-pm">PM1.0:</span> ${Math.round(
+				lastReading.pm01_0
+			)} μg/m³</p>
+      <p style="margin: 0; display: flex; justify-content: space-between"><span class="mapboxgl-pm">PM2.5:</span> ${Math.round(
+				lastReading.pm02_5
+			)} μg/m³</p>
+      <p style="margin: 0; display: flex; justify-content: space-between"><span class="mapboxgl-pm">PM10.0:</span> ${Math.round(
+				lastReading.pm10_0
+			)} μg/m³</p>`
+		);
+
+		markerColor = getAQIColor(lastReading.aqi);
+
+		airnoteMarker = new mapboxgl.Marker({ color: markerColor })
+			.setLngLat([lastReading.lon, lastReading.lat])
+			.setPopup(popup)
+			.addTo(map);
+
+		map.scrollZoom.disable();
+	});
+</script>
+
+{#if lastReading.lon && lastReading.lat}
+	<div in:fade>
```

Comment:
Since this page is server rendered now, I believe all these `in:fade` attributes aren’t doing anything anymore.

Reply:
Good call. I added some page transitions in instead to make switching routes a little smoother